app/main.py

```diff
@@ -2,0 +1,54 @@
+import os
+
+
+def move_file(command: str) -> None:
+    parts = command.split()
+
+    if len(parts) != 3 or parts[0] != "mv":
+        print("Error: Invalid command format.")
+        return
+
+    source_file, destination = parts[1], parts[2]
+
+    if not os.path.exists(source_file):
+        print(f"Error: {source_file} does not exist.")
+        return
+
+    if not os.path.isfile(source_file):
+        print(f"Error: {source_file} is a directory, not a file.")
+        return
+
+    if destination.endswith('/'):
+        destination_dir = destination
+        print(f"Creating directories: {destination_dir}")
+        try:
+            os.makedirs(destination_dir, exist_ok=True)
+        except FileExistsError:
+            print(f"Warning: {destination_dir} already exists.")
+        destination = os.path.join(destination_dir, os.path.basename(source_file))
+
+    else:
+        if os.path.exists(destination):
+            print(f"Error: {destination} already exists.")
+            return
+
+        destination_dir = os.path.dirname(destination)
+        if destination_dir and not os.path.exists(destination_dir):
+            print(f"Creating directories: {destination_dir}")
+            try:
+                os.makedirs(destination_dir, exist_ok=True)
+            except FileExistsError:
+                pass
+
+    try:
+        with open(source_file, "r") as src:
```

Comment:
Opening the source file in read mode ('r') assumes the file is a text file. If the file is binary, this will cause an error. Consider opening the file in binary mode ('rb') to handle both text and binary files.